nl_preprocessing.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import unicode_literals
+from functools import reduce
+from typing import Any
+
+from extractor import Function
+import re
+import nltk
+
+nltk.download('averaged_perceptron_tagger')
+nltk.download('stopwords')
+nltk.download('wordnet')
+
+# Precompile often used regex
+first_cap_regex = re.compile('(.)([A-Z][a-z]+)')
+all_cap_regex = re.compile('([a-z0-9])([A-Z])')
+
+
+class CleanFunction:
```

Comment:
I think it will cause maintenance issues if we to reimplement have code duplication from `Function` class. Also, I suspect we will add functionality to `Function` future. I suggest either extending the `Function` class _or_ adding a clean method to it.

Reply:
A `CleanFunction` represents something else than a `Function`, hence it does not make sense to mutate the fields of a `Function` object after preprocessing that field. `CleanFunction` is not a duplicate of `Function` and contains others functions etc. It's not like every function in `Function` should be available within `CleanFunction`, they are just separate classes. Thus, extending it does not really make sense.
Adding some fields to the original class for the cleaned version of the certain fields could be an option, but I'd rather treat them as immutable as we've done everywhere so far.

Reply:
The difference between a `Function` and `CleanFunction` is the removal of words in the natural language expressions right (function name, argument names etc)? 
If you want to keep the original `Function` immutable, then you could also create a new object with the cleaned up values.